### PR TITLE
EthereumLightClient: add crate eth-light-client 

### PR DESCRIPTION
**Pull request by shaymalichi**

### CODE REVIEW

File: crates/eth-light-client/Cargo.toml [hunk 1]
```diff
@@ -21,3 +21,6 @@ hex.workspace = true
 chrono.workspace = true
 byteorder.workspace = true
 bcs.workspace = true
+serde_json.workspace = true
+#sui-sdk = {path = "../sui-sdk"}
+#serde_json = "1.0.109"
```
Comment (shaymalichi) [inline crates/eth-light-client/Cargo.toml:26]:
```suggestion
```

**Review by zeevmoney — CHANGES_REQUESTED** [review]
This PR is not ready, base it on the correct branch.

File: crates/eth-light-client/src/config.rs [hunk 2]
```diff
@@ -0,0 +1,37 @@
+use helios::prelude::networks::Network;
```
Comment (zeevmoney) [inline crates/eth-light-client/src/config.rs:1]: Separate this to it's own PR

File: crates/eth-light-client/src/eth_state.rs [hunk 3]
```diff
@@ -0,0 +1,723 @@
...
+    pub time_slot: u64,
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct EthStateObject {
+    pub id: UID,
+    pub data: Vec<u8>,
+    pub time_slot: u64,
+}
+
```
Comment (zeevmoney) [inline crates/eth-light-client/src/eth_state.rs:51]: Separate this to it's own PR

File: Cargo.lock
Comment (zeevmoney) [inline Cargo.lock:1]: Why did you delete this file?

File: crates/eth-light-client/src/light_client.rs [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+use crate::proof::{Proof, ProofResponse, verify_proof};
+use crate::update::UpdatesResponse;
+use crate::utils;
+
+pub struct EthLightClient {
+    pub client: Client<FileDB>,
+    config: EthLightClientConfig,
+}
+
+impl EthLightClient {
```
Comment (zeevmoney) [inline crates/eth-light-client/src/light_client.rs:39]: Separate this to it's own PR + tests

File: crates/eth-light-client/src/proof.rs [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+/// Verifies the proof of a given storage value in a Merkle Patricia Tree.
+/// It takes a proof, storage root, path, and value as arguments and verifies the proof.
+/// The function checks for exclusion and inclusion proofs by iterating over each node in the proof.
+/// If the proof is valid, the function returns true. Otherwise, it returns false.
+/// # Arguments
+/// * `proof` - A reference to a slice of Bytes that represents the proof.
+/// * `root` - A reference to a slice of bytes that represents the root of the Merkle Patricia Tree (of the contract's storage).
+/// * `path` - A reference to a slice of bytes that represents the path to the value in the tree. it is built from hashing the key.
+/// * `value` - A reference to a Vec<u8> that represents the value to be verified.
+pub fn verify_proof(proof: &[Bytes], root: &[u8], path: &[u8], value: &Vec<u8>) -> bool {
```
Comment (zeevmoney) [inline crates/eth-light-client/src/proof.rs:35]: Separate to it's own PR +tests

File: crates/eth-light-client/src/update.rs [hunk 6]
```diff
@@ -0,0 +1,64 @@
+use helios::consensus::types::{
+    FinalityUpdate, FinalityUpdateSerde, OptimisticUpdate, OptimisticUpdateSerde, Update,
+    UpdateSerde,
+};
+use serde::{Deserializer, Serializer};
+
+#[derive(Debug, serde::Deserialize, serde::Serialize, Clone)]
```
Comment (zeevmoney) [inline crates/eth-light-client/src/update.rs:7]: Separate to it's own PR + tests

**Review by zeevmoney — CHANGES_REQUESTED** [review]
This should be separated to multiple PRs, see comments, closing.

**Pull request closed without merging**